src/odemis/acq/stitching/test/registrar_test.py

```diff
@@ -0,0 +1,533 @@
+            self.assertLessEqual(diff1, margin1,
+                                 "Failed for %s tiles, %s overlap and %s method," % (num, o, a) +
+                                 " %f != %f" % (tile_pos[i][0], pos[i][0]))
+            self.assertLessEqual(diff2, margin2,
+                                 "Failed for %s tiles, %s overlap and %s method," % (num, o, a) +
+                                 " %f != %f" % (tile_pos[i][1], pos[i][1]))
+
+        for i in range(len(dep_tile_pos)):
+            for j in range(len(dep_tile_pos[0])):
+
+                diff1 = abs(dep_tile_pos[i][j][0] - pos[j][0])
+                self.assertLessEqual(diff1, margin1,
+                                     "Failed for %s tiles, %s overlap and %s method," % (num, o, a) +
+                                     " %f != %f" % (dep_tile_pos[i][j][0], pos[i][0]))
+
+                self.assertLessEqual(diff2, margin2,
+                                     "Failed for %s tiles, %s overlap and %s method," % (num, o, a) +
+                                     " %f != %f" % (dep_tile_pos[i][j][1], pos[i][1]))
+
+        # Test with shifted dependent tiles
+        [tiles, pos] = decompose_image(img, o, num, a)
+
+        registrar = ShiftRegistrar()
+
+        # Add different shift for every dependent tile
+        dep_tiles = copy.deepcopy(tiles)
+        # x-shift for each dependent tile in px
+        rnd1 = [random.randrange(-1000, 1000) for _ in range(len(pos))]
+        # y-shift for each dependent tile in px
+        rnd2 = [random.randrange(-1000, 1000) for _ in range(len(pos))]
+        # Change metadata of dependent tiles
+        for i in range(len(dep_tiles)):
+            p = (dep_tiles[i].metadata[model.MD_POS][0] + rnd1[i] * px_size[0],
+                 dep_tiles[i].metadata[model.MD_POS][1] + rnd2[i] * px_size[1])
+            dep_tiles[i].metadata[model.MD_POS] = p
+
+        for i in range(len(pos)):
+            # Register tiles
+            # 2 layers of dependent tiles with the same pos
+            registrar.addTile(tiles[i], (dep_tiles[i], dep_tiles[i]))
+            tile_pos, dep_tile_pos = registrar.getPositions()
+
+            # Test main tile
+            diff1 = abs(tile_pos[i][0] - pos[i][0])
+            diff2 = abs(tile_pos[i][1] - pos[i][1])
+
+            # allow difference of 10% of overlap
+            margin1 = 0.1 * tiles[i].shape[0] * px_size[0]
+            margin2 = 0.1 * tiles[i].shape[1] * px_size[1]
+            self.assertLessEqual(diff1, margin1,
+                                 "Failed for %s tiles, %s overlap and %s method," % (num, o, a) +
+                                 " %f != %f" % (tile_pos[i][0], pos[i][0]))
+            self.assertLessEqual(diff2, margin2,
+                                 "Failed for %s tiles, %s overlap and %s method," % (num, o, a) +
+                                 " %f != %f" % (tile_pos[i][0], pos[i][0]))
+
+        # Test dependent tiles
+        for layer in dep_tile_pos:
```

Comment:
The position of the dependent tile is very precisely known: it's the position of the main_tile + the shift (aka rnd1/rnd2). So compute that expected value and use asserAlmostEqual to check it.